Zoltan2:  MultiJagged captures this in a lambda which breaks Clang 11 + Cuda

## Bug Report
@trilinos/zoltan2 

### Description
This is a non-standard build.  I built on ATS2 using IBM Clang (11) + Cuda directly (no NVCC wrapper).
When building Zoltan2 for MueLu, Clang falls over because MJ has `this->` in some of the lambda's.

A deeper look at the MJ functions shows one is a class `const` function and the other **nearly** a `const` function.

This makes the first a trivial candidate to move to a `static` class function (no this), and the second only uses `this->sEpislon`, which is pretty easy to hoist out and let that function be static as well.

Doing the priors, the code compiles and I can build Trilinos + Cuda with native Clang (which enables some other things I want to do!)

I've add a PR #9050 that fixes it (and hopefully doesn't break other builds...)

E.g.,
```
In file included from packages/muelu/src/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp:55:
In file included from /g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Rebalancing/MueLu_Zoltan2Interface_def.hpp:57:
In file included from /g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/problems/Zoltan2_PartitioningProblem.hpp:54:
In file included from /g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/partition/Zoltan2_PartitioningAlgorithms.hpp:55:
/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp:5185:29: error: capture host side class data member by this pointer in device or host device lambda function
        imbalance_on_left = calculate_imbalance(seen_weight_in_part,
                            ^
```

<details>
<summary> Full output </summary>

```
[1/55] Building CXX object packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp.o
FAILED: packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp.o 
/usr/tce/packages/mvapich2/mvapich2-2020.12.11-cuda-10.1.243-clang-ibm-11.0.1/bin/mpicxx -Dmuelu_EXPORTS -I. -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/BrickAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/Containers -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/CoupledAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/HybridAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/MatrixTransformation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/PairwiseAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/StructuredAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/StructuredAggregation/coupled -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/StructuredAggregation/uncoupled -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/UncoupledAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Graph/UserAggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Headers -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Interface -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Interface/FacadeClasses -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Misc -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/MueCentral -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Rebalancing -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Smoothers -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/BaseClass -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/BlackBox -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Smoothers/BlockedSmoothers -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/BlockedTransfers -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/Energy-Minimization -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/Energy-Minimization/Solvers -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/GeneralGeometric -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/Generic -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/Geometric-Interpolation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/Petrov-Galerkin-SA -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/SemiCoarsen -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/Smoothed-Aggregation -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/User -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Utils -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Utils/ForwardDeclaration -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/../adapters/xpetra -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/../adapters/tpetra -I/g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Transfers/PCoarsen -Ipackages/muelu/src -Ipackages/intrepid2/src -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Cell -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Discretization/Basis -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Discretization/FunctionSpaceTools -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Discretization/Integration -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Orientation -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Projection -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src/Shared -I/g/g20/jjellio/src/Trilinos-jje/packages/intrepid2/src -Ipackages/shards/src -I/g/g20/jjellio/src/Trilinos-jje/packages/shards/src -Ipackages/sacado/src -I/g/g20/jjellio/src/Trilinos-jje/packages/sacado/src -I/g/g20/jjellio/src/Trilinos-jje/packages/sacado/src/new_design -I/g/g20/jjellio/src/Trilinos-jje/packages/sacado/src/template -I/g/g20/jjellio/src/Trilinos-jje/packages/sacado/src/parameter -I/g/g20/jjellio/src/Trilinos-jje/packages/sacado/src/mpl -Ipackages/teuchos/kokkoscomm/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/kokkoscomm/src -Ipackages/teuchos/kokkoscompat/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/kokkoscompat/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/parameterlist/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/parser/src -Ipackages/teuchos/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/core/src -Ipackages/kokkos/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos/core/src -Ipackages/kokkos -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/comm/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/numerics/src -Ipackages/kokkos-kernels/src -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/impl -Ipackages/kokkos-kernels/src/impl -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/impl/tpls -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/blas -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/blas/impl -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/sparse -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/sparse/impl -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/graph -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/graph/impl -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/batched -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos-kernels/src/common -Ipackages/kokkos/algorithms/src -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos/algorithms/src -Ipackages/kokkos/containers/src -I/g/g20/jjellio/src/Trilinos-jje/packages/kokkos/containers/src -Ipackages/teko/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teko/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teko/src/NS -I/g/g20/jjellio/src/Trilinos-jje/packages/teko/src/Epetra -I/g/g20/jjellio/src/Trilinos-jje/packages/teko/src/Tpetra -Ipackages/stratimikos/src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/adapters/belos/src -Ipackages/stratimikos/adapters/belos/src -Ipackages/stratimikos/adapters/belos/src/../../../src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/adapters/amesos2/src -Ipackages/stratimikos/adapters/amesos2/src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/adapters/aztecoo/src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/adapters/amesos/src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/adapters/ml/src -I/g/g20/jjellio/src/Trilinos-jje/packages/stratimikos/adapters/ifpack/src -I/g/g20/jjellio/src/Trilinos-jje -Ipackages/amesos2/src -I/g/g20/jjellio/src/Trilinos-jje/packages/amesos2/src -I/g/g20/jjellio/src/Trilinos-jje/packages/amesos2/src/KLU2/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/amesos2/src/KLU2/Source -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/AMD/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/COLAMD/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/BTF/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/UFconfig -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/CAMD/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/CCOLAMD/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/common/auxiliarySoftware/SuiteSparse/src/KLU/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/core/ext -Ipackages/tpetra/core/ext -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/core/inout -Ipackages/tpetra/core/inout -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/core/src/kokkos_refactor -Ipackages/tpetra/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/tsqr/src -Ipackages/tpetra/tsqr/src -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/classic/LinAlg -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/classic/NodeAPI -Ipackages/tpetra/classic/NodeAPI -Ipackages/tpetra/classic/src -I/g/g20/jjellio/src/Trilinos-jje/packages/tpetra/classic/src -I/g/g20/jjellio/src/Trilinos-jje/packages/teuchos/remainder/src -Ipackages/teuchos/remainder/src -Ipackages/epetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/epetra/src -Ipackages/epetraext/src -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src/transform -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src/inout -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src/coloring -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src/model_evaluator -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src/block -I/g/g20/jjellio/src/Trilinos-jje/packages/epetraext/src/restrict -Ipackages/belos/xpetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/belos/xpetra/src -Ipackages/belos/tpetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/belos/tpetra/src -Ipackages/belos/epetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/belos/epetra/src -Ipackages/belos/src -I/g/g20/jjellio/src/Trilinos-jje/packages/belos/src -Ipackages/aztecoo/src -I/g/g20/jjellio/src/Trilinos-jje/packages/aztecoo/src -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/sup/Matrix -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/sup/StridedMap -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/sup/Utils -Ipackages/xpetra/sup -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/CrsGraph -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/CrsMatrix -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/BlockedCrsMatrix -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/DistObject -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Export -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Headers -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Import -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Map -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/BlockedMap -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/MultiVector -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/BlockedMultiVector -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/BlockedVector -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Operator -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Platform -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/RowGraph -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/RowMatrix -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Utils -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Utils/ForwardDeclaration -I/g/g20/jjellio/src/Trilinos-jje/packages/xpetra/src/Vector -Ipackages/xpetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/adapters/tpetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/adapters/epetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/interfaces/operator_vector/fundamental -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/interfaces/operator_vector/extended -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/support/operator_vector/client_support -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/support/operator_vector/adapter_support -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/interfaces/operator_solve/fundamental -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/interfaces/operator_solve/extended -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/support/operator_solve/client_support -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/interfaces/nonlinear/model_evaluator/fundamental -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/support/nonlinear/model_evaluator/client_support -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/interfaces/nonlinear/solvers/fundamental -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/src/support/nonlinear/solvers/client_support -Ipackages/thyra/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/core/example/operator_vector -I/g/g20/jjellio/src/Trilinos-jje/packages/rtop/src -I/g/g20/jjellio/src/Trilinos-jje/packages/rtop/src/interfaces -I/g/g20/jjellio/src/Trilinos-jje/packages/rtop/src/support -I/g/g20/jjellio/src/Trilinos-jje/packages/rtop/src/ops_lib -I/g/g20/jjellio/src/Trilinos-jje/packages/rtop/src/lapack -Ipackages/rtop/src -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/adapters/epetraext/src/model_evaluator -I/g/g20/jjellio/src/Trilinos-jje/packages/thyra/adapters/epetraext/src/transformer -Ipackages/ml/src -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Include -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Comm -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Coarsen -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/FEGrid -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Main -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Smoother -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Krylov -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Operator -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/Utils -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/RefMaxwell -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/MatrixFree -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/MLAPI -I/g/g20/jjellio/src/Trilinos-jje/packages/ml/src/LevelWrap -Ipackages/ifpack/src -I/g/g20/jjellio/src/Trilinos-jje/packages/ifpack/src -Ipackages/amesos/src -I/g/g20/jjellio/src/Trilinos-jje/packages/amesos/src -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-xpetra -Ipackages/galeri/src-xpetra -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-xpetra/../src-epetra -Ipackages/galeri/src-xpetra/../src-epetra -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-xpetra/../src/Utils -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-xpetra/../src/Headers -Ipackages/galeri/src-epetra -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-epetra -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-epetra/Maps -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-epetra/CRSMatrices -I/g/g20/jjellio/src/Trilinos-jje/packages/galeri/src-epetra/FiniteElements -Ipackages/zoltan/src -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/include -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/all -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/coloring -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/graph -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/ha -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/hier -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/hsfc -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/lb -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/matrix -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/order -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/par -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/params -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/tpls -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/phg -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/rcb -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/reftree -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/simple -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/timer -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/Utilities/Communication -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/Utilities/DDirectory -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/Utilities/Timer -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/Utilities/shared -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan/src/zz -I/g/g20/jjellio/src/Trilinos-jje/packages/ifpack2/adapters/thyra -Ipackages/ifpack2/adapters -I/g/g20/jjellio/src/Trilinos-jje/packages/ifpack2/src -Ipackages/ifpack2/src -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src -Ipackages/zoltan2/core/src -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/partition -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/order -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/color -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/match -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/map -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/zoltan -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/environment -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/include -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/input -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/models -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/problems -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/TpetraCrsColorer -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/util -I/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/directory -Ipackages/anasazi/tpetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/anasazi/tpetra/src -Ipackages/anasazi/epetra/util/ModeLaplace -I/g/g20/jjellio/src/Trilinos-jje/packages/anasazi/epetra/util/ModeLaplace -Ipackages/anasazi/epetra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/anasazi/epetra/src -Ipackages/anasazi/src -I/g/g20/jjellio/src/Trilinos-jje/packages/anasazi/src -I/g/g20/jjellio/src/Trilinos-jje/packages/anasazi/src/../thyra/src -I/g/g20/jjellio/src/Trilinos-jje/packages/anasazi/src/../tpetra/src -mcpu=power9 -mtune=power9  -x cuda --cuda-gpu-arch=sm_70  -O3 -DNDEBUG -fPIC -std=c++14 -MD -MT packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp.o -MF packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp.o.d -o packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp.o -c packages/muelu/src/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp
In file included from packages/muelu/src/Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp:55:
In file included from /g/g20/jjellio/src/Trilinos-jje/packages/muelu/src/Rebalancing/MueLu_Zoltan2Interface_def.hpp:57:
In file included from /g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/problems/Zoltan2_PartitioningProblem.hpp:54:
In file included from /g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/partition/Zoltan2_PartitioningAlgorithms.hpp:55:
/g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp:5185:29: error: capture host side class data member by this pointer in device or host device lambda function
        imbalance_on_left = calculate_imbalance(seen_weight_in_part,
                            ^
/g/g20/jjellio/src/Trilinos-jje/packages/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel.hpp:679:5: note: called by 'exec_team<void>'
    m_functor(member);
    ^
/g/g20/jjellio/src/Trilinos-jje/packages/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel.hpp:725:22: note: called by 'operator()'
      this->template exec_team<WorkTag>(typename Policy::member_type(
                     ^
/g/g20/jjellio/src/Trilinos-jje/packages/kokkos/core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp:106:3: note: called by 'cuda_parallel_launch_constant_memory<Kokkos::Impl::ParallelFor<(lambda at /g/g20/jjellio/src/Trilinos-jje/packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp:5128:41), Kokkos::TeamPolicy<Kokkos::Cuda>, Kokkos::Cuda>>'
  driver();
  ^
1 error generated when compiling for sm_70.
```
</details>

### Steps to Reproduce
1. SHA1:  [insert here]
1. Configure script:  [attach here]
1. Configure log:  [attach here]
1. Build log:  [attach here]
1. Input deck:  [attach here]
1. Do this.
1. Do that.
1. Shake fist angrily at computer.
1. Run log:  [attach here]

--- a/packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp
+++ b/packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp
@@ -645,8 +645,9 @@ private:
    * \param achieved balance we achieved.
    * \param expected balance expected.
    */
+  static
   KOKKOS_INLINE_FUNCTION
-  double calculate_imbalance(mj_scalar_t achieved, mj_scalar_t expected) const {
+  double calculate_imbalance(mj_scalar_t achieved, mj_scalar_t expected) {
     return static_cast<double>(achieved) / static_cast<double>(expected) - 1.0;
   }
 
@@ -715,6 +716,7 @@ private:
    * the left of the cut line.
    * \param new_cut_position DOCWORK: Documentation
    */
+  static
   KOKKOS_INLINE_FUNCTION
   void mj_calculate_new_cut_position (
     mj_scalar_t cut_upper_bound,
@@ -722,7 +724,8 @@ private:
     mj_scalar_t cut_upper_weight,
     mj_scalar_t cut_lower_weight,
     mj_scalar_t expected_weight,
-    mj_scalar_t &new_cut_position);
+    mj_scalar_t &new_cut_position,
+    mj_scalar_t sEpsilon);
 
   /*! \brief Function checks if should do migration or not.
    * It returns true to point that migration should be done when
@@ -4386,19 +4389,21 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t, mj_node_t>::
  */
 template <typename mj_scalar_t, typename mj_lno_t, typename mj_gno_t,
   typename mj_part_t, typename mj_node_t>
+KOKKOS_INLINE_FUNCTION
 void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
   mj_node_t>::mj_calculate_new_cut_position(mj_scalar_t cut_upper_bound,
   mj_scalar_t cut_lower_bound,
   mj_scalar_t cut_upper_weight,
   mj_scalar_t cut_lower_weight,
   mj_scalar_t expected_weight,
-  mj_scalar_t &new_cut_position) {
+  mj_scalar_t &new_cut_position,
+  mj_scalar_t sEpsilon) {
 
-  if(std::abs(cut_upper_bound - cut_lower_bound) < this->sEpsilon) {
+  if(std::abs(cut_upper_bound - cut_lower_bound) < sEpsilon) {
     new_cut_position = cut_upper_bound; //or lower bound does not matter.
   }
 
-  if(std::abs(cut_upper_weight - cut_lower_weight) < this->sEpsilon) {
+  if(std::abs(cut_upper_weight - cut_lower_weight) < sEpsilon) {
     new_cut_position = cut_lower_bound;
   }
 
@@ -5113,6 +5118,7 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
   auto local_global_min_max_coord_total_weight =
     global_min_max_coord_total_weight;
 
+  const auto _sEpsilon = this->sEpsilon;
   // Note for a 22 part system I tried removing the outer loop
   // and doing each sub loop as a simple parallel_for over num_cuts.
   // But that was about twice as slow (10ms) as the current form (5ms)
@@ -5154,6 +5160,8 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
 
     Kokkos::parallel_for(Kokkos::TeamThreadRange (team_member, num_cuts),
       [=] (mj_part_t i) {
+      using algMJ_t = AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
+                            mj_node_t>;
       // seen weight in the part
       mj_scalar_t seen_weight_in_part = 0;
       // expected weight for part.
@@ -5180,12 +5188,12 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
         //expected ratio
         expected_weight_in_part = current_part_target_weights(i);
 
-       //leftImbalance = imbalanceOf(seenW, globalTotalWeight, expected);
-        imbalance_on_left = calculate_imbalance(seen_weight_in_part,
+        //leftImbalance = imbalanceOf(seenW, globalTotalWeight, expected);
+        imbalance_on_left = algMJ_t::calculate_imbalance(seen_weight_in_part,
           expected_weight_in_part);
         // rightImbalance = imbalanceOf(globalTotalWeight - seenW,
         // globalTotalWeight, 1 - expected);
-        imbalance_on_right = calculate_imbalance(global_total_weight -
+        imbalance_on_right = algMJ_t::calculate_imbalance(global_total_weight -
           seen_weight_in_part, global_total_weight - expected_weight_in_part);
         bool is_left_imbalance_valid = std::abs(imbalance_on_left) -
           used_imbalance_tolerance < local_sEpsilon ;
@@ -5301,12 +5309,13 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
             }
 
             mj_scalar_t new_cut_position = 0;
-            this->mj_calculate_new_cut_position(
+            algMJ_t::mj_calculate_new_cut_position(
               current_cut_upper_bounds(i),
               current_cut_lower_bounds(i),
               current_cut_upper_weights(i),
               current_cut_lower_bound_weights(i),
-              expected_weight_in_part, new_cut_position);
+              expected_weight_in_part, new_cut_position,
+              _sEpsilon);
 
             // if cut line does not move significantly.
             // then finalize the search.
@@ -5382,13 +5391,14 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
             }
           }
           mj_scalar_t new_cut_position = 0;
-          this->mj_calculate_new_cut_position(
+          algMJ_t::mj_calculate_new_cut_position(
             current_cut_upper_bounds(i),
             current_cut_lower_bounds(i),
             current_cut_upper_weights(i),
             current_cut_lower_bound_weights(i),
             expected_weight_in_part,
-            new_cut_position);
+            new_cut_position,
+            _sEpsilon);
 
             // if cut line does not move significantly.
             if(std::abs(current_cut_coordinates(i) -

